### DISCUSSION TITLE
Revalidate all requests

### DISCUSSION THREAD

**Discussion by JuanCaicedo** [vercel/swr#984] [Q&A]
Hi everyone! I'm using SWR to show a paginated component, very similar to the [example in the docs](https://swr.vercel.app/docs/pagination#pagination). My key looks something like 
```
useSWR(`/api/items?offset=${offset}`)
```

I would like to upload a new list of items, and I want to tell SWR to revalidate all items requests since they will no longer be valid. I'm assuming `mutate` is still the right avenue to do this, but is there a way that I can tell mutate to invalidate many keys at once? Or ever better, all keys that match `/api/items(.*)`?

Thank you! I've also asked this on Stackoverflow if any of you would like to answer there 😀 https://stackoverflow.com/questions/66231471/how-to-mutate-multiple-swr-keys

**Comment by ErnestBrandi** [comment]
If `offset` is a known number, I guess you can just iterate over api keys and manually revalidate (https://github.com/vercel/swr#manually-revalidate).

**Reply by JuanCaicedo** [reply]
`offset` depends on the current page the user is looking at, so I don't think iterating through all of them would be a robust solution here 😅

If the user views 30 pages, then I would need to revalidate `/api/items?offset=0`, `/api/items?offset=10`, and `/api/items?offset=20`. If they view more pages, then I would need to revalidate even more 😕

**Reply by ErnestBrandi** [reply]
> If they view more pages, then I would need to revalidate even more

I am not sure this is a bad thing; and as far as my understanding goes this is exactly what `useSWRInfinite` does. 
Also keep in mind that SWR has by default a `revalidateOnFocus` flag set to `true`; which means anytime your browser window is focused again, all your page data is revalidated anyway.



> I would like to upload a new list of items, and I want to tell SWR to revalidate all items requests since they will no longer be valid

Considering this, revalidating those pages doesn't make much sens neither as you might have a different number of pages after all. If you use a code similar to the one in your page example :
```
for (let i = 0; i < cnt; i++) {
    pages.push(<Page index={i} key={i} />)
}
```
then you will have a different set of pages with their own hooks and you won't have to bother revalidating stuffs.
